pkgs/tools/system/lshw/default.nix

```diff
@@ -1,27 +1,33 @@
-{ stdenv, fetchurl
-, withGUI? false, gtk? null, pkgconfig? null, sqlite? null  # compile GUI
- }:
-stdenv.mkDerivation rec {
+{ stdenv, lib, fetchurl, fetchpatch
+, withGUI ? false, gtk2, pkgconfig, sqlite # compile GUI
```

Comment:
Somewhat unintuitively, `callPackage` fills in parameters that are `null` by default , so they'll still be passed in unless you explicitly set them to null in `all-packages.nix`.

But perhaps you're thinking of making it easier to invoke the expression without `callPackage`?